main/src/ca/uwaterloo/flix/language/ast/MonoType.scala

```diff
@@ -69,7 +69,7 @@ object MonoType {
   // Compound Types.
   //
 
-  case class Array(tpe: MonoType) extends MonoType
+  case class Array(tpe: MonoType, purity: Purity) extends MonoType
```

Comment:
What????

Reply:
Looks crazy but its really just exposing how it is (we monomorphize regions into IO)